src/service/data-util.service.ts

```diff
@@ -124,15 +125,21 @@ export class JhiDataUtils {
         return size.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ' ') + ' bytes';
     }
 
-    setFileData(event, entity, field: string, isImage: boolean) {
+    setFileData(event, entity, field: string, isImage: boolean, cb?: Function) {
         if (event && event.target.files && event.target.files[0]) {
             const file = event.target.files[0];
             if (isImage && !/^image\//.test(file.type)) {
+                if (cb) {
+                    cb();
```

Comment:
shouldn't this return an error message to caller?

Reply:
previous implementation doesn't return anything.
My change was just to allow accepting and executing a callback passed by the caller

Reply:
sorry to be not explicit, but, was referring about passing error/result to the callback function (error as first arg and result as second arg). It's up to consumer to handle or ignore outcome of such functions. This is general convention used in `node.js` as well.

On further thinking, another solution could be to return `promise` as response and resolve/reject promises for success/failure scenario. Considering current code is executed asynchronously (that's why need for callback is felt), I don't see this as breaking change.

Reply:
Ah - makes a lot of sense now. I do like the idea of returning a promise because it might make testing easier.

However I'm not entirely sure how we can achieve that since another callback (passed to toBase64()) needs to complete before we return any promise that resolves to the modified entity object.

That said I can, minimally, pass the entity to the callback (as your proposed option 1) in both situations.
I'll update the PR soon to reflect this.

Reply:
@vishal423, I just realized (after looking at https://github.com/jhipster/ng-jhipster/pull/82), that passing 2 callbacks would make more sense, since the 2 places where the setFileData() exits represent 2 separate scenarios:
1. a successful completion and 
2. an erroneous completion 

of the function.

If you agree, I can update my PR to accommodate this change and update setFileData to accept 2 callbacks

Reply:
I was thinking inline to below:

```
    setFileData(event, entity, field: string, isImage: boolean): Promise<any> {
        return new Promise((resolve, reject) => {
            if (event && event.target.files && event.target.files[0]) {
                const file = event.target.files[0];
                if (isImage && !/^image\//.test(file.type)) {
                    reject('invalid image content type');
                }
                this.toBase64(file, (base64Data) => {
                    entity[field] = base64Data;
                    entity[`${field}ContentType`] = file.type;
                    resolve(entity);
                });
            } else {
                reject('no files selected');
            }
        });
    }
```

Reply:
@vivekmore I'm ok with passing two callbacks (onSuccess, onError)

Reply:
thanks @wmarques - I've updated the PR to accept 2 callbacks

Reply:
@vishal423, forgive me if I'm missing something, but if we return a promise wouldn't that be a breaking change? 

Nothing will get executed unless/until the caller calls .then() on the returned promise

Reply:
@vivekmore, I think not as js promises are guaranteed to be executed.

Reply:
@vishal423 , I stand corrected. Thank you for pointing that out.
I'll modify the code to accommodate your suggestion and remove the callback parameters.